### ISSUE TITLE
commits incorrectly dissapearing when using `gps pull`

### ISSUE THREAD

**Issue by drewdeponte** [uptech/git-ps-rs#43]
Basically, everything seems good up until I run the `gps pull` command at which point it properly drops the commit that it should. But it also is dropping the "Add garbage.rs" commit incorrectly for some reason.

```
~/code/uptech/git-ps-test
✔ gps ls                                                                                                                                                                                        12m main 83ab0e0
5           83ab0e Add foopty to hoopty.rs
4           2c3044 Add hoopty.rs
3           7f5ba6 Add foo()
2           9212c6 Add your mom message
1           d7dba2 Change hello world message to Hello, bob
0           b325ad Add garbage.rs

~/code/uptech/git-ps-test
✔ git la                                                                                                                                                                                        13m main 83ab0e0
* 83ab0e0 - G - (HEAD -> main, old-main) Add foopty to hoopty.rs (13 minutes ago) <Drew De Ponte>
* 2c3044d - G - Add hoopty.rs (9 weeks ago) <Drew De Ponte>
* 7f5ba68 - G - Add foo() (3 months ago) <Drew De Ponte>
* 9212c6b - G - Add your mom message (3 months ago) <Drew De Ponte>
* d7dba20 - G - Change hello world message to Hello, bob (3 months ago) <Drew De Ponte>
* b325ad6 - N - Add garbage.rs (9 weeks ago) <Drew De Ponte>
* a34b62b - G - (origin/main) Initial skeleton (7 months ago) <Drew De Ponte>

~/code/uptech/git-ps-test
✔ ../git-ps-rs/target/debug/gps ls                                                                                                                                                              13m main 83ab0e0
5           83ab0e Add foopty to hoopty.rs
4           2c3044 Add hoopty.rs
3           7f5ba6 Add foo()
2           9212c6 Add your mom message
1           d7dba2 Change hello world message to Hello, bob
0           b325ad Add garbage.rs

~/code/uptech/git-ps-test
✔ ../git-ps-rs/target/debug/gps sync 4                                                                                                                                                          15m main 83ab0e0
Enumerating objects: 6, done.
Counting objects: 100% (6/6), done.
Delta compression using up to 12 threads
Compressing objects: 100% (3/3), done.
Writing objects: 100% (4/4), 425 bytes | 425.00 KiB/s, done.
Total 4 (delta 1), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (1/1), completed with 1 local object.
remote:
remote: Create a pull request for 'ps/rr/add_hoopty_rs' on GitHub by visiting:
remote:      https://github.com/uptech/git-ps-test/pull/new/ps/rr/add_hoopty_rs
remote:
To github.com:uptech/git-ps-test.git
 * [new branch]      ps/rr/add_hoopty_rs -> ps/rr/add_hoopty_rs

~/code/uptech/git-ps-test
✔ git la                                                                                                                                                                                        15m main 83ab0e0
* 83ab0e0 - G - (HEAD -> main, old-main) Add foopty to hoopty.rs (15 minutes ago) <Drew De Ponte>
* 2c3044d - G - Add hoopty.rs (9 weeks ago) <Drew De Ponte>
* 7f5ba68 - G - Add foo() (3 months ago) <Drew De Ponte>
* 9212c6b - G - Add your mom message (3 months ago) <Drew De Ponte>
* d7dba20 - G - Change hello world message to Hello, bob (3 months ago) <Drew De Ponte>
* b325ad6 - N - Add garbage.rs (9 weeks ago) <Drew De Ponte>
| * 2fc4684 - N - (origin/ps/rr/add_hoopty_rs, ps/rr/add_hoopty_rs) Add hoopty.rs (9 weeks ago) <Drew De Ponte>
|/
* a34b62b - G - (origin/main) Initial skeleton (7 months ago) <Drew De Ponte>

~/code/uptech/git-ps-test
✔ ../git-ps-rs/target/debug/gps int 4                                                                                                                                                           15m main 83ab0e0
merge_base_oid = a34b62b61d1675073eb1df953f1cdfc01232cceb
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
To github.com:uptech/git-ps-test.git
   a34b62b..2fc4684  origin/ps/rr/add_hoopty_rs -> main
To github.com:uptech/git-ps-test.git
 - [deleted]         ps/rr/add_hoopty_rs

~/code/uptech/git-ps-test
✔ git la                                                                                                                                                                                        15m main 83ab0e0
* 83ab0e0 - G - (HEAD -> main, old-main) Add foopty to hoopty.rs (16 minutes ago) <Drew De Ponte>
* 2c3044d - G - Add hoopty.rs (9 weeks ago) <Drew De Ponte>
* 7f5ba68 - G - Add foo() (3 months ago) <Drew De Ponte>
* 9212c6b - G - Add your mom message (3 months ago) <Drew De Ponte>
* d7dba20 - G - Change hello world message to Hello, bob (3 months ago) <Drew De Ponte>
* b325ad6 - N - Add garbage.rs (9 weeks ago) <Drew De Ponte>
| * 2fc4684 - N - (origin/main) Add hoopty.rs (9 weeks ago) <Drew De Ponte>
|/
* a34b62b - G - Initial skeleton (7 months ago) <Drew De Ponte>

~/code/uptech/git-ps-test
✔ git pull --rebase                                                                                                                                                                             16m main 83ab0e0
dropping 2c3044d849d8de1b19779a0368e34c969ebdbed1 Add hoopty.rs -- patch contents already upstream
Successfully rebased and updated refs/heads/main.

~/code/uptech/git-ps-test
✔ git la                                                                                                                                                                                        18m main 6f803d2
* 6f803d2 - G - (HEAD -> main) Add foopty to hoopty.rs (18 minutes ago) <Drew De Ponte>
* b512e42 - G - Add foo() (3 months ago) <Drew De Ponte>
* 647087d - G - Add your mom message (3 months ago) <Drew De Ponte>
* 0571a2e - G - Change hello world message to Hello, bob (3 months ago) <Drew De Ponte>
* 2fc4684 - N - (origin/main) Add hoopty.rs (9 weeks ago) <Drew De Ponte>
| * 83ab0e0 - G - (old-main) Add foopty to hoopty.rs (18 minutes ago) <Drew De Ponte>
| * 2c3044d - G - Add hoopty.rs (9 weeks ago) <Drew De Ponte>
| * 7f5ba68 - G - Add foo() (3 months ago) <Drew De Ponte>
| * 9212c6b - G - Add your mom message (3 months ago) <Drew De Ponte>
| * d7dba20 - G - Change hello world message to Hello, bob (3 months ago) <Drew De Ponte>
| * b325ad6 - N - Add garbage.rs (9 weeks ago) <Drew De Ponte>
|/
* a34b62b - G - Initial skeleton (7 months ago) <Drew De Ponte>

~/code/uptech/git-ps-test
✔ git push -f origin a34b62b:main                                                                                                                                                               18m main 6f803d2
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
To github.com:uptech/git-ps-test.git
 + 2fc4684...a34b62b a34b62b -> main (forced update)
```

**Comment by drewdeponte** [comment]
Given that the `gps pull` command is currently implemented as simply `git pull --rebase` I decided to vet that the problem was actually with the `git pull --rebase`. Actually the output above is output that shows that the problem is happening with `git pull --rebase`.

So I tested doing a `git rebase --onto origin/main origin/main main` which is what I did in the previous incarnation of `git-ps`. It works correctly, drops the patch that was integrated and no other commits.

So I think I just need to implement that again for the `gps pull` command rather than using the `git pull --rebase`.

It is worth noting that I dug into the docs around `git pull --rebase` to see if I could find some option or something to control how it does the rebase so that it would drop commits it shouldn't. Sadly I couldn't find anything.

So again the right thing to do is to implement `git rebase --onto origin/main origin/main main`.
